app/models/main.py

```diff
@@ -1832,6 +1848,302 @@ def serialize(self):
+                        "awardingOrganisationName": self.awarding_organisation_name,
+                        # don't risk json representing this as a float
+                        "awardValue": str(self.award_value) if self.award_value is not None else None,
+                    },
+                } if self.result == "awarded" else {}
+            ),
+        }
+
+    @validates(
+        "start_date",
+        "end_date",
+        "awarding_organisation_name",
+        "award_value",
+    )
+    def _validates_data_complete_if_completed_at(self, key, value):
+        if key == "award_value":
+            if value is not None and value < 0:
+                raise ValidationError(f"{key} must be greater than or equal to zero, got {value!r}")
+
+        if self.completed_at and self.result == "awarded" and value in (None, "",):
+            raise ValidationError(
+                f"{key} cannot be None or empty if {self.result!r} ProcessOutcome is 'completed'. Received {value!r}"
+            )
+
+        if self.result not in (None, "awarded",) and value is not None:
+            raise ValidationError(
+                f"{key} cannot be set for {self.result!r} ProcessOutcomes. Attempted to set value {value!r}"
+            )
+
+        return value
+
+    @validates("completed_at", "result")
+    def _validates_completed_at_data_complete_if_set(self, key, value):
+        if key == "result" and value not in self.RESULT_CHOICES:
+            raise ValidationError(f"{value!r} is not a valid choice for field {key!r}")
+
+        result = value if key == "result" else self.result
+        completed_at = value if key == "completed_at" else self.completed_at
+        if result == "awarded" and completed_at is not None:
+            failures = ", ".join(
+                "{} = {}".format(fkey, repr(getattr(self, fkey)))
+                for fkey in (
+                    "start_date",
+                    "end_date",
+                    "awarding_organisation_name",
+                    "award_value",
+                )
+                if (getattr(self, fkey) in (None, "",))
+            )
+            if failures:
+                raise ValidationError(
+                    f"{result!r} ProcessOutcome cannot be 'completed' with None or empty data fields, but {failures}"
+                )
+        if result not in (None, "awarded",):
+            failures = ", ".join(
+                "{} = {}".format(fkey, repr(getattr(self, fkey)))
+                for fkey in (
+                    "start_date",
+                    "end_date",
+                    "awarding_organisation_name",
+                    "award_value",
+                )
+                if getattr(self, fkey) is not None
+            )
+            if failures:
+                raise ValidationError(
+                    f"{result!r} ProcessOutcomes cannot have award-related data fields set, but {failures}"
```

Comment:
This feels like it might read weirdly in the `None` case, don't really know if we should care tho.

Reply:
Actually, these will never get shown in the `None` case.

Reply:
OK.

I don't think the validation messages should include the model names either - they should be couched in terms of the API.

Reply:
But of course, if you look at it, these error messages are also referring to the *model* field names, rather than the API field names, so the man behind the curtain is in fact in full view :tornado:

Reply:
Agh trying to reformulate all these messages but finding something common that makes sense is hard - then again note how rarely these error messages make it out of the endpoint.